### ISSUE TITLE
pip install -U turicreate Showing Error (using python3 and windows 10)

### ISSUE THREAD

**Issue by hridoy100** [apple/turicreate#3149]
Collecting turicreate
  Using cached turicreate-6.0.tar.gz (1.9 kB)
Building wheels for collected packages: turicreate
  Building wheel for turicreate (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\Hridoy\Anaconda3\envs\virtual_environment_name\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Hridoy\\AppData\\Local\\Temp\\pip-install-yglwdhwi\\turicreate\\setup.py'"'"'; __file__='"'"'C:\\Users\\Hridoy\\AppData\\Local\\Temp\\pip-install-yglwdhwi\\turicreate\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\Hridoy\AppData\Local\Temp\pip-wheel-yqhedwiv'
       cwd: C:\Users\Hridoy\AppData\Local\Temp\pip-install-yglwdhwi\turicreate\
  Complete output (31 lines):
  running bdist_wheel
  running build
  installing to build\bdist.win-amd64\wheel
  running install

          ==================================================================================
          TURICREATE ERROR

          If you see this message, pip install did not find an available binary package
          for your system.

          Supported Platforms:
              * macOS 10.12+ x86_64.
              * Linux x86_64 (including WSL on Windows 10).

          Support Python Versions:
              * 2.7
              * 3.5
              * 3.6
              * 3.7

          Another possible cause of this error is an outdated pip version. Try:
              `pip install -U pip`

          ==================================================================================

  ----------------------------------------
  ERROR: Failed building wheel for turicreate
  Running setup.py clean for turicreate
Failed to build turicreate
Installing collected packages: turicreate
    Running setup.py install for turicreate ... error
    ERROR: Command errored out with exit status 1:
     command: 'C:\Users\Hridoy\Anaconda3\envs\virtual_environment_name\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Hridoy\\AppData\\Local\\Temp\\pip-install-yglwdhwi\\turicreate\\setup.py'"'"'; __file__='"'"'C:\\Users\\Hridoy\\AppData\\Local\\Temp\\pip-install-yglwdhwi\\turicreate\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Hridoy\AppData\Local\Temp\pip-record-d_6jd0af\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\Hridoy\Anaconda3\envs\virtual_environment_name\Include\turicreate'
         cwd: C:\Users\Hridoy\AppData\Local\Temp\pip-install-yglwdhwi\turicreate\
    Complete output (28 lines):
    running install

            ==================================================================================
            TURICREATE ERROR

            If you see this message, pip install did not find an available binary package
            for your system.

            Supported Platforms:
                * macOS 10.12+ x86_64.
                * Linux x86_64 (including WSL on Windows 10).

            Support Python Versions:
                * 2.7
                * 3.5
                * 3.6
                * 3.7

            Another possible cause of this error is an outdated pip version. Try:
                `pip install -U pip`

            ==================================================================================

    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Users\Hridoy\Anaconda3\envs\virtual_environment_name\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Hridoy\\AppData\\Local\\Temp\\pip-install-yglwdhwi\\turicreate\\setup.py'"'"'; __file__='"'"'C:\\Users\\Hridoy\\AppData\\Local\\Temp\\pip-install-yglwdhwi\\turicreate\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Hridoy\AppData\Local\Temp\pip-record-d_6jd0af\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\Hridoy\Anaconda3\envs\virtual_environment_name\Include\turicreate' Check the logs for full command output.

**Comment by TobyRoseman** [comment]
What version of Python 3 are you using? I suspect you are using Python 3.8 which we don't support yet (see #3099).

**Comment by ABHISHEK-T-S** [comment]
use wsl ,turicreate is not  supported in windows

**Comment by tssrao** [comment]
I have also got the same error. I have even reinstalled the windows. Installed Python 3.7 first. Then, I have installed the WSL and then tried to install turicreate. Someone please help me to install turicreate in my Windows PC
![Screenshot (6)](https://user-images.githubusercontent.com/64457083/82073368-c6d21780-96f6-11ea-947f-a5c249cc6371.png)

**Comment by ABHISHEK-T-S** [comment]
MAYBE YOU COULD USE GRAPHLAB
IT WORK WELL ON WINDOWS

**Comment by nishit5629** [comment]
I have got the same problem ,i am using ubuntu(20.16) and python 3.8.
after scrolling allot over the internet i finally   find nothing helpful , please help

**Comment by ABHISHEK-T-S** [comment]
it is not supported in python 3.8 yet

**Comment by tssrao** [comment]
After scrolling through many discussions and forums, I have found the solution.
Use UBUNTU 18.04 and python 3.7
Any higher versions of both will not support .
This will definitely solve the problem.

**Comment by bhavi-san** [comment]
@tssrao did it work for you?

**Comment by tssrao** [comment]
@bhavi-san, It is working absolutely fine.

**Comment by bhavi-san** [comment]
alright thanks

**Comment by Manish-Akkapalli** [comment]
Thanks @tssrao it totally works

**Comment by Madhu296** [comment]
> 
> 
> @bhavi-san, It is working absolutely fine.

What was the command used to install turicreate in virtualenv? I am getting the error

**Comment by tssrao** [comment]
@Madhu296, share the screenshot of your error.

**Comment by Madhu296** [comment]
Dear sir,

Please find the attached screenshot of my error page during turicreate
installation.
Thanks in advance.

Madhu

**Comment by Madhu296** [comment]
![Turicreate error](https://user-images.githubusercontent.com/67204699/85216366-58414300-b3a1-11ea-8468-3967b211c262.PNG)

**Comment by tssrao** [comment]
Which version of Ubuntu you are using?

**Comment by Madhu296** [comment]
python 3.7.7 and ubuntu 18.4, still turicreate is not installing for me

**Comment by Madhu296** [comment]
ubuntu 18.04 LTS

**Comment by Jahnavi34** [comment]
I am using python version 3.7 and ubuntu 18.04 LTS version. But I am still getting the same error.Can anyone please help??

**Comment by tssrao** [comment]
@Jahnavi34, error screenshot?

**Comment by amitnandan** [comment]
i am getting a tensor flow error while installing turicreate. Could anyone 
help..
![error](https://user-images.githubusercontent.com/65026421/85770297-6217cd00-b738-11ea-80f8-dbc11a62829e.PNG)

**Comment by TobyRoseman** [comment]
@amitnandan - try upgrading your version of `pip`. You can do this by running: `pip install -U pip`. Please try that, then try to install TuriCreate again.

**Comment by Jahnavi34** [comment]
> @Jahnavi34, error screenshot?

I am actually trying to install Turicreate using Ubuntu 18.04 LTS and i have installed it using pip3 in virtual environment.
Then when i am trying to import turicreate in Jupyter notebook it is showing an error **stating**moduke not found****
please help me

**Comment by ABHISHEK-T-S** [comment]
looks like your pip and your jupyter notebook are using different python environments.

**Comment by ABHISHEK-T-S** [comment]
> > @Jahnavi34, error screenshot?
> 
> I am actually trying to install Turicreate using Ubuntu 18.04 LTS and i have installed it using pip3 in virtual environment.
> Then when i am trying to import turicreate in Jupyter notebook it is showing an error stating module not found
> please help me
> ![Screenshot (28)](https://user-images.githubusercontent.com/67423376/85832719-32a3a780-b7ae-11ea-9d00-e70f37609714.png)

looks like your pip and your jupyter notebook are using different python environments.

**Comment by Jenithareddy** [comment]
used pip:

Collecting turicreate
  Could not fetch URL https://pypi.python.org/simple/turicreate/: There was a problem confirming the ssl certificate: [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:661) - skipping
  Could not find a version that satisfies the requirement turicreate (from versions: )
No matching distribution found for turicreate

used pip3:

Collecting turicreate
  Using cached turicreate-6.0.tar.gz (1.9 kB)
Using legacy setup.py install for turicreate, since package 'wheel' is not installed.
Installing collected packages: turicreate
    Running setup.py install for turicreate ... error
    ERROR: Command errored out with exit status 1:
     command: /Library/Frameworks/Python.framework/Versions/3.8/bin/python3.8 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/_9/lbklj4n517bdkh5z68w7fd100000gn/T/pip-install-b232jirz/turicreate/setup.py'"'"'; __file__='"'"'/private/var/folders/_9/lbklj4n517bdkh5z68w7fd100000gn/T/pip-install-b232jirz/turicreate/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/_9/lbklj4n517bdkh5z68w7fd100000gn/T/pip-record-tr4i1z28/install-record.txt --single-version-externally-managed --compile --install-headers /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/turicreate
         cwd: /private/var/folders/_9/lbklj4n517bdkh5z68w7fd100000gn/T/pip-install-b232jirz/turicreate/
    Complete output (28 lines):
    running install
    
    
    
            ==================================================================================
            TURICREATE ERROR
    
            If you see this message, pip install did not find an available binary package
            for your system.
    
            Supported Platforms:
                * macOS 10.12+ x86_64.
                * Linux x86_64 (including WSL on Windows 10).
    
            Support Python Versions:
                * 2.7
                * 3.5
                * 3.6
                * 3.7
    
    
            Another possible cause of this error is an outdated pip version. Try:
                `pip install -U pip`
    
            ==================================================================================
    
    
    
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Library/Frameworks/Python.framework/Versions/3.8/bin/python3.8 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/_9/lbklj4n517bdkh5z68w7fd100000gn/T/pip-install-b232jirz/turicreate/setup.py'"'"'; __file__='"'"'/private/var/folders/_9/lbklj4n517bdkh5z68w7fd100000gn/T/pip-install-b232jirz/turicreate/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/_9/lbklj4n517bdkh5z68w7fd100000gn/T/pip-record-tr4i1z28/install-record.txt --single-version-externally-managed --compile --install-headers /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/turicreate Check the logs for full command output.



I am getting this error too. I tried to install in macOS 10.15.2 and python version Python 2.7.14

**Comment by TobyRoseman** [comment]
@Jenithareddy - Looking at the output you gave, the path to your Python executable is `/Library/Frameworks/Python.framework/Versions/3.8/bin/python3.8`. That means you are using Python 3.8 not Python 2.7. TuriCreate does not yet support Python 3.8 (we're tracking this in #3099). If you use python 2.7 with macOS 10.15, it should work.

**Comment by priyanka3598** [comment]
If you see this message, pip install did not find an available binary package
          for your system.

          Supported Platforms:
              * macOS 10.12+ x86_64.
              * Linux x86_64 (including WSL on Windows 10).

          Support Python Versions:
              * 2.7
              * 3.5
              * 3.6
              * 3.7

iam getting error like this my python version is 3.7 not 3.8 and i also enabled my windows via wsl even then it is not getting installed

**Comment by priyanka3598** [comment]
can you please help me with this!!

**Comment by tssrao** [comment]
@priyanka3598 
I have seen many users facing problems while installing turicreate. I have spent two days to get it done.I thought that it will be my last try to install turicreate and if I still fail to install it, I will give up the course. I got it after several ways of trying. Read the below instructions to get turicreate into your Windows PC.

Before starting the installation of turicreate, read all the instructions first, and then start the installing process ensure the following:

- Your PC is running on Windows 10

- Uninstall Python3.8/any newer versions.

- Install Python 3.7

- While Installing Python 3.7, make sure that you click the add to path checkbox.

Now its time to install turicreate.

Follow each step one by one in this link: https://medium.com/@pam.thumbela/installing-turicreate-on-windows-10-534e147a4792

While going through the steps in the above link, make sure that:

- You open the Windows Powershell as an administrator at step-1.

- At step-3, install Ubuntu 18.04 LTS from Microsoft store. Don't install Ubuntu 20.04 LTS. This is where most people do mistake and find it difficult to install because in the last step it will throw an error if you use Ubuntu 20.04 LTS.

- Remember that whenever you want to work with turicreate, you should always open jupyter notebook from ubuntu itslef in the venv(virtual environment). Don't open jupyter notebook from command prompt.

If you are still facing issues, reply to this with the screenshot of the error you are getting. I will try to help you.

If you still face issues regarding this, write to tsaishodhanrao@gmail.com. I will try to answer your issues.

**Comment by tssrao** [comment]
@priyanka3598 ,
ubuntu 18.04 LTS should be always running in the background while running jupyter notebbok and it should be in the virtual environment(venv).

**Comment by priyanka3598** [comment]
Thankyou for your reply. Will try it

**Comment by Gnaneswri** [comment]
I was not able to install turicreate ,i tried graph lab also. im getting no module found. can anyone help me

**Comment by jaswantcoder** [comment]
> I have also got the same error. I have even reinstalled the windows. Installed Python 3.7 first. Then, I have installed the WSL and then tried to install turicreate. Someone please help me to install turicreate in my Windows PC
> ![Screenshot (6)](https://user-images.githubusercontent.com/64457083/82073368-c6d21780-96f6-11ea-947f-a5c249cc6371.png)

I am also getting exactly same error. Yes, I have installed and set python version 3.7 and I am trying in WSL

**Comment by Nishit-KKPandit** [comment]
Try with anaconda3

**Comment by tssrao** [comment]
@Gnaneswri , @jaswantcoder ,
Please attach screenshots of your errors...
and please make sure that you are using Python 3.7 and Ubuntu 18.04LTS

**Comment by Krejdom** [comment]
So this package is NOT compatible with Windows 10?

**Comment by TobyRoseman** [comment]
> So this package is NOT compatible with Windows 10?

It should be compatible with Windows 10, if you use WSL Ubuntu 18, not Ubuntu 20.

**Comment by ankur290109** [comment]
Getting below error while installing Turicreate, I have installed Python 3.8 and Anaconda 3:-

ERROR: Failed building wheel for turicreate
Running setup.py clean for turicreate
Failed to build turicreate
Installing collected packages: turicreate
Running setup.py install for turicreate ... error
ERROR: Command errored out with exit status 1:
command: 'C:\Users\Abhishek\anaconda3\envs\virtual_environment_name\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\Users\Abhishek\AppData\Local\Temp\pip-install-qhqi5iee\turicreate\setup.py'"'"'; file='"'"'C:\Users\Abhishek\AppData\Local\Temp\pip-install-qhqi5iee\turicreate\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(file);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, file, '"'"'exec'"'"'))' install --record 'C:\Users\Abhishek\AppData\Local\Temp\pip-record-92_w9hxx\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\Abhishek\anaconda3\envs\virtual_environment_name\Include\turicreate'
cwd: C:\Users\Abhishek\AppData\Local\Temp\pip-install-qhqi5iee\turicreate
Complete output (28 lines):
running install

        ==================================================================================
        TURICREATE ERROR

        If you see this message, pip install did not find an available binary package
        for your system.

        Supported Platforms:
            * macOS 10.12+ x86_64.
            * Linux x86_64 (including WSL on Windows 10).

        Support Python Versions:
            * 2.7
            * 3.5
            * 3.6
            * 3.7
            * 3.8

        Another possible cause of this error is an outdated pip version. Try:
            `pip install -U pip`

        ==================================================================================



----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Users\Abhishek\anaconda3\envs\virtual_environment_name\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\Users\Abhishek\AppData\Local\Temp\pip-install-qhqi5iee\turicreate\setup.py'"'"'; file='"'"'C:\Users\Abhishek\AppData\Local\Temp\pip-install-qhqi5iee\turicreate\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(file);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, file, '"'"'exec'"'"'))' install --record 'C:\Users\Abhishek\AppData\Local\Temp\pip-record-92_w9hxx\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\Abhishek\anaconda3\envs\virtual_environment_name\Include\turicreate' Check the logs for full command output.

**Comment by TobyRoseman** [comment]
@ankur290109 - rather than posting the same message in three different existing issue, please open a new issue. Also please include a lot more details, such as your operating system, what Linux distro you are using with WSL, etc.

**Comment by TobyRoseman** [comment]
It is now clear to me that the original issue reported by @hridoy100 was the result of trying to install TuriCreate on Window outside of WSL. It was not a Python 3.8 issue as I originally thought. We are already tracking native Windows support in #533. So the original issue here is a duplicate.

There has been a large amount of correspondence in this issue. Other people may have gotten this error message mentioned here for a different reason. If you got this error message on Windows **and** used WSL, please [open a new issue](https://github.com/apple/turicreate/issues/new/choose). 

I'm going to close this issue as a duplicate.

Thank you @ABHISHEK-T-S and @tssrao for your help with this issue.
